tsdb/db.go

```diff
@@ -597,6 +597,85 @@ func (db *DBReadOnly) Blocks() ([]BlockReader, error) {
 	return blockReaders, nil
 }
 
+// LastBlockID returns the BlockID of latest block.
+func (db *DBReadOnly) LastBlockID(logger log.Logger) (string, error) {
+	select {
+	case <-db.closed:
+		return "", ErrClosed
+	default:
+	}
+	latestDirName, err := db.lastBlockDirName(logger)
+	if err != nil {
+		return "", err
+	}
+	// Open the latest block and get its ID
+	block, err := db.Block(logger, latestDirName)
+	if err != nil {
+		return "", err
+	}
+	return block.Meta().ULID.String(), nil
+}
+
+func (db *DBReadOnly) lastBlockDirName(logger log.Logger) (string, error) {
+	entries, err := os.ReadDir(db.dir)
+	if err != nil {
+		return "", err
+	}
+	skipNames := map[string]struct{}{
+		"index":          {},
+		"chunks_head":    {},
+		"lock":           {},
+		"queries.active": {},
+		"wal":            {},
+	}
+	max := uint64(0)
+
+	latestDirName := ""
+	// Walk the blocks directory and find the latest subdirectory
+	for _, e := range entries {
+		dirName := e.Name()
+
+		if _, skip := skipNames[dirName]; skip {
+			// Skip the  directory
+			continue
+		}
+
+		ulidObj, err := ulid.Parse(dirName)
+		if err != nil {
+			return "", err
+		}
+		timestamp := ulidObj.Time()
+
+		if timestamp > max {
+			max = timestamp
+			latestDirName = dirName
+		}
+	}
+
+	if latestDirName == "" {
+		return "", errors.New("no blocks found")
+	}
+
+	return latestDirName, nil
+}
+
+// Block returns a block reader by given block id.
+func (db *DBReadOnly) Block(logger log.Logger, blockID string) (BlockReader, error) {
```

Comment:
the `db` object already has a logger. We don't need to pass it here.

Reply:
I agree that since the db object already has a logger configured, we can use it directly without passing a logger object to the method. I have made necessary changes to remove the logger parameter and utilize the logger associated with the db object